tests/v2/validation/provisioning/k3s/README.md

```diff
@@ -8,118 +8,129 @@ Please see below for more details for your config. Please see below for more det
 
 ## Table of Contents
 1. [Prerequisites](../README.md)
-2. [Define your test](#provisioning-input)
-3. [Cloud Credential](#cloud-credentials)
-4. [Configure providers to use for Node Driver Clusters](#machine-k3s-config)
-5. [Configuring Custom Clusters](#custom-cluster)
-6. [Static test cases](#static-test-cases) 
-7. [Advanced Cluster Settings](#advanced-settings)
-8. [Back to general provisioning](../README.md)
+2. [Configuring test flags](#Flags)
+3. [Define your test](#provisioning-input)
+4. [Cloud Credential](#cloud-credentials)
+5. [Configure providers to use for Node Driver Clusters](#machine-k3s-config)
+6. [Configuring Custom Clusters](#custom-cluster)
+7. [Static test cases](#static-test-cases)
+8. [Advanced Cluster Settings](#advanced-settings)
+9. [Back to general provisioning](../README.md)
+
+## Flags
+Flags are used to determine which static table tests are run (has no effect on dynamic tests) 
+`Long` Will run the long version of the table tests (usually all of them)
+`Short` Will run the subset of table tests with the short flag.
+
+```yaml
+flags:
+  desiredflags: "Long"   #required (static tests only)
+```
 
 ## Provisioning Input
-provisioningInput is needed to the run the K3S tests, specifically kubernetesVersion and providers. nodesAndRoles is only needed for the TestProvisioningDynamicInput test, node pools are divided by "{nodepool},". psact is optional and takes values `rancher-privileged`, `rancher-restricted` or `rancher-baseline`.
+provisioningInput is needed to the run the K3S tests, specifically kubernetesVersion and providers.
 
 **nodeProviders is only needed for custom cluster tests; the framework only supports custom clusters through aws/ec2 instances.**
 ```yaml
 provisioningInput:
   machinePools:
-  - nodeRoles:
-      etcd: true
+  - nodeRoles:                                              #required(dynamic only) (at least 1)
+      etcd: true                                            #required(dynamic only) (at least 1 controlplane & etcd & worker)
       controlplane: true
       worker: true
       quantity: 1
-  - nodeRoles:
+  - nodeRoles:                  
       worker: true
       quantity: 2
       drainBeforeDelete: true
   - nodeRoles:
+      worker: true
       windows: true
       quantity: 1
-  flags:
-    desiredflags: "Long" #These flags are for running TestProvisioningK3SCluster or TestProvisioningK3SCustomCluster it is not needed for the dynamic tests. Long will run the full table, where as short will run the short version of this test.
-  rke2KubernetesVersion: ["v1.26.8+k3s1"]
-  providers: ["linode", "aws", "do", "harvester"]
-  nodeProviders: ["ec2"]
+  rke2KubernetesVersion: ["v1.27.6+k3s1"]                    #required (at least 1)                                             
+  providers: ["linode", "aws", "do", "harvester"]            #required (at least 1)
```

Comment:
is this list supposed to be complete? If so, we should add azure, vsphere, google, maybe others if I missed any.

Reply:
I would actually put the ones we mainly use. i.e. linode, aws, azure, vsphere.